examples/run_mosaic_trainer.py

```diff
@@ -12,33 +12,26 @@
     --algorithms label_smoothing --alpha 0.1
     --datadir ~/datasets
 """
-import argparse
-import logging
+import sys
+import warnings
+from typing import Type
 
-import composer
 from composer.trainer.trainer import Trainer
 from composer.trainer.trainer_hparams import TrainerHparams
 
-logger = logging.getLogger(__name__)
+
+def warning_on_one_line(message: str, category: Type[Warning], filename: str, lineno: int, file=None, line=None):
+    # From https://stackoverflow.com/questions/26430861/make-pythons-warnings-warn-not-mention-itself
+    return f'{category.__name__}: {message} (source: {filename}:{lineno})\n'
 
 
 def main() -> None:
-    logging.basicConfig()
-    logging.captureWarnings(True)
+    warnings.formatwarning = warning_on_one_line
 
-    parser = argparse.ArgumentParser(parents=[TrainerHparams.get_argparse(cli_args=True)])
```

Comment:
This was unnecessary the whole time? 😳

Reply:
Nah, only after we moved the `datadir` to trainer hparams.